### ISSUE TITLE
Audiofile (.mp3) showing up as a subfolder and not a file.

### ISSUE THREAD

**Issue by nford88** [izderadicka/audioserve#48]
Hey there, enjoying the software, its exactly what I need.

I have come across an issue with a few files though. For some reason, I have a mp3 file which keeps displaying as a subfolder. I thought it had an extra stream in, when I click on the .mp3 name it then takes me to another sub directory where I can play the actual mp3 file and a random non-playable file.

It should just play my mp3 file like the others. I have included a few snapshots below. Most other MP3 files work fine. Also included is an output of ffprobe with two files, one of which displays as a file, the other as a sub-folder. Only difference I can see is the encoder tag. Any help would be great.

![image](https://user-images.githubusercontent.com/19550724/110128777-189b7980-7dbf-11eb-8032-98aa62d32fc2.png)

<img width="1410" alt="image" src="https://user-images.githubusercontent.com/19550724/110128479-bd698700-7dbe-11eb-8a76-9ad665021050.png">

ffprobe output for working file
```json{
    "streams": [
        {
            "index": 0,
            "codec_name": "mp3",
            "codec_long_name": "MP3 (MPEG audio layer 3)",
            "codec_type": "audio",
            "codec_time_base": "1/44100",
            "codec_tag_string": "[0][0][0][0]",
            "codec_tag": "0x0000",
            "sample_fmt": "fltp",
            "sample_rate": "44100",
            "channels": 2,
            "channel_layout": "stereo",
            "bits_per_sample": 0,
            "r_frame_rate": "0/0",
            "avg_frame_rate": "0/0",
            "time_base": "1/14112000",
            "start_pts": 353600,
            "start_time": "0.025057",
            "duration_ts": 11389501440,
            "duration": "807.079184",
            "bit_rate": "128000",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0,
                "timed_thumbnails": 0
            },
            "tags": {
                "encoder": "LAME3.96r"
            },
            "side_data_list": [
                {
                    "side_data_type": "Replay Gain"
                }
            ]
        }
    ],
    "format": {
        "filename": "Sylvia Plath  - The Bell Jar CD1 - Track 01.mp3",
        "nb_streams": 1,
        "nb_programs": 0,
        "format_name": "mp3",
        "format_long_name": "MP2/3 (MPEG audio layer 2/3)",
        "start_time": "0.025057",
        "duration": "807.079184",
        "size": "13027567",
        "bit_rate": "129132",
        "probe_score": 51,
        "tags": {
            "disc": "1/7",
            "track": "1/38"
        }
    }
}```

ffprobe output for file showing as a subfolder
```json{
    "streams": [
        {
            "index": 0,
            "codec_name": "mp3",
            "codec_long_name": "MP3 (MPEG audio layer 3)",
            "codec_type": "audio",
            "codec_time_base": "1/44100",
            "codec_tag_string": "[0][0][0][0]",
            "codec_tag": "0x0000",
            "sample_fmt": "fltp",
            "sample_rate": "44100",
            "channels": 2,
            "channel_layout": "stereo",
            "bits_per_sample": 0,
            "r_frame_rate": "0/0",
            "avg_frame_rate": "0/0",
            "time_base": "1/14112000",
            "start_pts": 353600,
            "start_time": "0.025057",
            "duration_ts": 9596805120,
            "duration": "680.045714",
            "bit_rate": "128000",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0,
                "timed_thumbnails": 0
            },
            "tags": {
                "encoder": "Lavc58.91"
            }
        }
    ],
    "format": {
        "filename": "Part 0 - Intro.mp3",
        "nb_streams": 1,
        "nb_programs": 0,
        "format_name": "mp3",
        "format_long_name": "MP2/3 (MPEG audio layer 2/3)",
        "start_time": "0.025057",
        "duration": "680.045714",
        "size": "10885633",
        "bit_rate": "128057",
        "probe_score": 51,
        "tags": {
            "major_brand": "isom",
            "minor_version": "512",
            "compatible_brands": "isomiso2mp41",
            "comment": "",
            "copyright": "Penguin Random House Audio Publishing Group",
            "media_type": "2",
            "genre": "African American Nonfiction/Biography & Autobiography/Nonfiction/Politics",
            "disc": "0",
            "track": "1",
            "title": "Part 0 - Intro",
            "artist": "Barack Obama",
            "album": "A Promised Land",
            "album_artist": "A Promised Land",
            "encoder": "Lavf58.45.100"
        }
    }
}```

**Comment by izderadicka** [comment]
Thanks for detailed description of the problem.   

audioserve is using libavformat (from ffmpeg package) to extract audio meta data. 
If audio file contains chapters then it is displayed as "pseudo" folder - so you can access it's chapters individually.   It's used mostly for .m4b format, which contains chapters information.
How ever I've never seen chapters info in .mp3 file.    (audioserve can split huge mp3 file into equal size parts - chapter like - but it has to be enabled by explicit option -  or one can supply additional  companion file to indicate chapters  -   I assume you are not using this).

So I think libavformat is returning chapters for this file  - something like one real and one dummy?

Can you share the problematic file?

**Comment by nford88** [comment]
It seems you are right, I couldn't find that anywhere but of my full metadata output using ffmpeg of my file does show it has 2 chapters there for some reason.

This was originally a chapters m4b file which I split using the m4b-tool utility. 
[Ironically I found the maintainer of that post the exact issue I had whereby you cannot seem to re-import new chapters data from a modified metadata file](https://trac.ffmpeg.org/ticket/8728).

Will look at trying to re-encode somehow.

**Comment by izderadicka** [comment]
Thanks for update -   anyhow if you can share file I'll look into it. 
m4b format should work with some limitation (no cover, no description).
I should add some sanity checking about chapters metadata ( if there is only 1 chapter do not show as folder etc..)

For splitting m4b I'm using [my own wrapper around ffmpeg](https://github.com/izderadicka/split_audiobook)

**Comment by nford88** [comment]
I also came up with a short bash script to strip out the chapters but retain all other metadata. It now appearing as it should 👍 Do you still want the file for your own observations

```bash
#!/bin/bash

# creates a direectory scrubbed with new chapoter removed files
# '.strip_chapters.sh <dir>

cd "$1"
mkdir "scrubbed"
set -x
for FILE in *; 
do 
   if [ ${FILE: -4} == ".mp3" ]
      then
         ffmpeg -i "${FILE}" -f ffmetadata "meta.txt"
         ffmpeg -i "$FILE" -i "meta.txt" -map_metadata 1 -map_chapters -1 -codec copy "./scrubbed/${FILE}"
         rm -rf "meta.txt"
   fi
done
```

👏 
<img width="436" alt="image" src="https://user-images.githubusercontent.com/19550724/110208741-dcc8e880-7e80-11eb-9c03-c4d68f699c62.png">

**Comment by izderadicka** [comment]
Yes please - share the file, I have made small changes - wondering if it helps.
Thnx

**Comment by nford88** [comment]
Here you go! -**https://filebin.net/gxqpu8fcce5dha7d**

**Comment by izderadicka** [comment]
Thanks -   I've checked - with ffprobe - there are really two chapters - first one is "real" - other has zero length (and starts and ends at end of first chapter) .  audioserve  displays them both and plays OK. So I think problem is in splitting program, which adds second useless  chapter.   I've added option `--ignore-chapters-meta`  to globally ignore chapters metadata if people would prefer that.
